jax/api.py

```diff
@@ -1966,6 +1966,40 @@ def device_put(x, device: Optional[xc.Device] = None):
   return tree_map(lambda y: xla.device_put_p.bind(y, device=device), x)
 
 
+def device_put_sharded(x, devices: Sequence[xc.Device]):
+  """Shards the input to specified devices, returning ShardedDeviceArrays.
+
+  Args:
+    x: A sequence of arrays, scalars, or (nested) standard Python containers thereof
+    devices: A sequence of devices()
+
+  Returns:
+    A ShardedDeviceArray or (nested) Python container thereof containing a copy
+    of x sharded across the specified devices.
+  """
```

Comment:
The flax replicate function has `x = jax.numpy.asarray(x)`. The other versions don't have this, so I'm not sure it's necessary, but you might wanna ask about it or include it to be safe (maybe it's to handle replicating scalars? Does device_put not handle that?)

Reply:
That's right - I think to exactly replicate the various forms of this that are used by downstream libraries will require some extra wrappers along these lines.

Other implementations pass nested containers to `x`, so calling `jnp.asarray(x)` would not work in all cases.